state_test.go

```diff
@@ -318,6 +318,25 @@ func TestStateUpdateReportStatus(t *testing.T) {
 	assert.WithinDuration(t, now2, time.Now(), 3*time.Second)
 
 	maxReportSendingTime = old
+
+	// pretend update was aborted at the backend, but was applied
+	// successfully on the device
+	usr = NewUpdateStatusReportState(update, client.StatusSuccess)
+	sc = &stateTestController{
+		reportError: NewFatalError(client.ErrDeploymentAborted),
+	}
+	s, c = usr.Handle(&ctx, sc)
+	assert.IsType(t, s, &ReportErrorState{})
+	assert.Equal(t, client.StatusSuccess, s.(*ReportErrorState).status)
```

Comment:
This mixing of error and success also confuses me a bit. It signals that the server has reported an error. But why is the status "success"? I mean, I get that it is because we updated successfully, but why is the success code inside the error state?